packages/schemas/src/types/system.ts

```diff
@@ -64,6 +64,42 @@ export const storageProviderGuard: Readonly<{
   [StorageProviderKey.StorageProvider]: storageProviderDataGuard,
 });
 
+// Email service provider
+export enum EmailServiceProvider {
+  SendGrid = 'SendGrid',
+}
+
+export const emailServiceDataGuard = z.discriminatedUnion('provider', [
+  z.object({
+    provider: z.literal(EmailServiceProvider.SendGrid),
```

Comment:
Can we use `z.enum` to define the enum and use it in the `provider` field, instead of using `literal SendGrid`?

Reply:
the `provider` field works as a discriminator, should use literal type. Can check other discriminated union type for reference.